client/payment-methods/index.js

```diff
@@ -83,30 +146,59 @@ const PaymentMethods = () => {
 		);
 	};
 
+	const {
+		featureFlags: { upe: initialUPESetting },
+	} = useContext( WCPaySettingsContext );
+	const [ isUpeEnabled, setIsUpeEnabled ] = useIsUpeEnabled(
+		initialUPESetting
+	);
+	const { status } = useContext( WcPayUpeContext );
```

Comment:
At this point using the hook is not necessary in this case, you can just do 
```suggestion
	const { isUpeEnabled, setIsUpeEnabled, status } = useContext(
		WcPayUpeContext
	);
```

`useIsUpeEnabled` does not accept an argument, so `useContext( WCPaySettingsContext )` can be removed as well.

Reply:
The trouble here is that I don't see how `WcPayUpeContext` can get the initial value for the UPE flag stored in the site's actual settings. It's only wired up to update the flag. Grabbing the flag from `WCPaySettingsContext` tells us on page load if it's enabled within the site. Otherwise, I could enable it, save settings, reload, and it would still be disabled.

Reply:
It should be passed here, when the context provider is mounted: https://github.com/Automattic/woocommerce-payments/blob/tweak/payment-methods-section/client/settings/settings-manager/index.js#L108
Similarly to how you did it in the tests: https://github.com/Automattic/woocommerce-payments/blob/tweak/payment-methods-section/client/payment-methods/test/index.js#L41

The `useIsUpeEnabled` function/hook does not accept an argument

Reply:
Updated.

>The useIsUpeEnabled function/hook does not accept an argument

I dunno what the hell I was thinking there 🤦, I think I was trying to set the initial value within the Provider but did it wrong.